sdk/core/switcher/entry.S

```diff
@@ -786,6 +789,12 @@ exception_entry_asm:
 	// Call the handler.
 	cjalr              cra
 
+	/*
+	 * Now that we're back, defer interrupts again before we do anything that
+	 * manipulates the TrustedStack.
+	 */
+	csrci              mstatus, 0x8
+
```

Comment:
Is this necessary?  I believe it's fine to take interrupts here and we'll just handle the sentry on the way back.

Reply:
I wasn't sufficiently sure to do anything else.  While seemingly simple, this chunk of code connects to the rest of the switcher rather deeply... The "handler asked for a context install" path is probably fine, but force unwind gets us to the cross-call return path, which presently runs with IRQs deferred, and that can get us to the thread-exit path, which *definitely* needs to run with IRQs deferred, since it zeros MTDC.

Would you be OK with this instruction in place for now and a post #320 cleanup of IRQ dispositions?

Reply:
I hit approve, so I'm happy as is (it isn't worse than what we have elsewhere), would be nice to have a comment suggesting that this might be safe to remove.